File: frontend/lib/auth/password-policy.ts
// Política de complejidad en cliente (espejo del backend PasswordPolicy, HU #10171):
// mínimo 8 caracteres con al menos una mayúscula, una minúscula y un dígito.
export const PASSWORD_MIN_LENGTH = 8;

export const PASSWORD_POLICY_HINT =
  "Mínimo 8 caracteres, con al menos una mayúscula, una minúscula y un número.";

export function isPasswordCompliant(password: string, minLength = PASSWORD_MIN_LENGTH): boolean {
  if (!password || password.length < minLength) {
    return false;
  }
  return /[A-Z]/.test(password) && /[a-z]/.test(password) && /\d/.test(password);
}

File: services/core-api/src/Flit.Modules.Security.Domain/Auth/PasswordPolicy.cs
namespace Flit.Modules.Security.Domain.Auth;

/// <summary>
/// Política de complejidad de contraseñas (HU #10171, RNF02): longitud mínima y al menos
/// una mayúscula, una minúscula y un dígito.
/// </summary>
public static class PasswordPolicy
{
    public const int DefaultMinLength = 8;

    public static bool IsCompliant(string password, int minLength = DefaultMinLength)
    {
        if (string.IsNullOrEmpty(password) || password.Length < minLength)
            return false;

        bool hasUpper = false, hasLower = false, hasDigit = false;
        foreach (var c in password)
        {
            if (char.IsUpper(c)) hasUpper = true;
            else if (char.IsLower(c)) hasLower = true;
            else if (char.IsDigit(c)) hasDigit = true;
        }

        return hasUpper && hasLower && hasDigit;
    }
}
